packages/core/src/state-management/repository.ts

```diff
@@ -614,10 +614,16 @@ export class Repository {
         return false
       }
       case AnchorRequestStatusName.COMPLETED: {
-        if (anchorEvent.cid.equals(state$.tip)) {
+        const anchorCommitCID = anchorEvent.witnessCar.roots[0]
+        await this._handleAnchorCommit(state$, anchorEvent.cid, anchorEvent.witnessCar)
+        if (state$.tip.equals(anchorCommitCID)) {
```

Comment:
Do we have a `if anchorHistory.includes(anchorCommitCID)`?

Reply:
depends on what you mean by "anchorHistory".  We can check if the anchor commit CID exists anywhere in the log of the state, but if we use that as the criteria to delete the entry from the anchorRequestStore, we risk deleting a valid entry for a newer update